doc/guide/pyapi.md

```diff
@@ -0,0 +1,493 @@
+SLING frame store concepts.
+
+SLING frames live in a store, so you create a new global store this way:
+```
+import sling
+commons = sling.Store()
+```
+Loading frames into the store:
+```
+commons.load("data/nlp/schemas/meta-schema.sling")
+commons.load("data/nlp/schemas/document-schema.sling")
+```
+Loading binary encoded frames in legacy encoding:
+```
+actions = commons.load("local/sempar/out/table", binary=True)
+```
+Freezing a store to makes it read-only and allows local stores to be created based on the store:
+```
+commons.freeze()
+```
+Looking up frames in the store:
+```
+name = commons['name']
+```
+Create a local store:
+```
+store = sling.Store(commons)
+```
+Frames in the global store are now accessible from the local store:
+```
+doc = store['/s/document']
+```
+Role values for frames can be accessed as attributes:
+```
+print doc.name
+```
+or using indexing:
+```
+print doc['name']
+```
+You can also use a frame value to access roles:
+```
+print doc[name]
+```
+You can test if a frame has a role:
+```
+if 'name' in doc: print "doc has 'name'"
+if name in doc: print "doc has name"
+```
+The `parse()` method can be used for adding new frames to the store:
+```
+f = store.parse('{a:10 b:10.5 c:"hello" d:{:/s/thing} e:[1,2,3]}')
+```
+The `frame()` method can be used to create a new frame from a dictionary:
+```
+f = store.frame({'a': 10, 'b': 10.5, 'c': "hello"})
+```
+or a list of slot tuples:
+```
+f = store.frame([('a', 10), ('b': 10.5), ('c': "hello")])
+```
+Slots can be added or modified using attribute assignment:
+```
+f.c = "hello world"
+```
+or using index assignment:
+```
+f[name] = "The F frame"
+f['a'] = 20
+```
+New slots can be added using the `append()` method:
+```
+f.d.append(name, "A thing")
+```
+Multiple slot can be added using the `extend()` method:
+```
+f.extend({'foo': 10, 'bar': 20})
+```
+or using a list of slot tuples:
+```
+f.extend([('foo', 10), ('bar': 20)])
+```
+All the slots in a frame can be iterated:
+```
+for name, value in f:
+  print "slot", name,"=", value
+```
+or just the roles with a particular name:
+```
+for r in doc('role'):
+  print "doc role", r
+```
+Frames can be encoded in text format with the `data()` method:
+```
+print f.data()
+```
+and with indentation:
+```
+print f.data(pretty=True)
+```
+or with binary encoding:
+```
+print len(f.data(binary=True))
+```
+Arrays can be created with the `array()` method:
+```
+a = store.array([1, 2, 3])
+```
+Arrays can also be created with nil values that can be assigned later:
+```
+a = store.array(3)
+a[0] = 1
+a[1] = 2
+a[2] = 3
+```
+SLING arrays work much in the same way as Python lists except that they have
+a fixed size:
+```
+print len(a)
+print a[1]
+for item in a: print item
+```
+Finally, a store can be save to a file in textual encoding:
+```
+store.save("/tmp/txt.sling")
+```
+or in binary encoding:
+```
+store.save("/tmp/bin.sling", binary=True)
+```
+
+## Phrase tables
```

Comment:
I think a natural progression from frames would be to recordio/documents instead of phrase tables. How about moving this to before Miscellaneous.